omnigibson/robots/r1.py

```diff
@@ -0,0 +1,519 @@
+        self._world_base_fixed_joint_prim = None
+
+        # Run super init
+        super().__init__(
+            relative_prim_path=relative_prim_path,
+            name=name,
+            uuid=uuid,
+            scale=scale,
+            visible=visible,
+            fixed_base=True,
+            visual_only=visual_only,
+            self_collisions=self_collisions,
+            load_config=load_config,
+            abilities=abilities,
+            control_freq=control_freq,
+            controller_config=controller_config,
+            action_type=action_type,
+            action_normalize=action_normalize,
+            reset_joint_pos=reset_joint_pos,
+            obs_modalities=obs_modalities,
+            proprio_obs=proprio_obs,
+            sensor_config=sensor_config,
+            grasping_mode=grasping_mode,
+            disable_grasp_handling=disable_grasp_handling,
+            **kwargs,
+        )
+
+    @property
+    def discrete_action_list(self):
+        # Not supported for this robot
+        raise NotImplementedError()
+
+    def _create_discrete_action_space(self):
+        # Fetch does not support discrete actions
+        raise ValueError("R1 does not support discrete actions!")
+
+    def _initialize(self):
+        # Run super method first
+        super()._initialize()
+
+    def reset(self):
+        """
+        Reset should not change the robot base pose.
+        We need to cache and restore the base joints to the world.
+        """
+        base_joint_positions = self.get_joint_positions()[self.base_idx]
+        super().reset()
+        self.set_joint_positions(base_joint_positions, indices=self.base_idx)
+
+    def _post_load(self):
+        super()._post_load()
+        # The eef gripper links should be visual-only. They only contain a "ghost" box volume for detecting objects
+        # inside the gripper, in order to activate attachments (AG for Cloths).
+        for arm in self.arm_names:
+            self.eef_links[arm].visual_only = True
+            self.eef_links[arm].visible = False
+
+        self._world_base_fixed_joint_prim = lazy.omni.isaac.core.utils.prims.get_prim_at_path(
+            f"{self.prim_path}/rootJoint"
+        )
+        position, orientation = self.get_position_orientation()
+        # Set the world-to-base fixed joint to be at the robot's current pose
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localPos0").Set(tuple(position))
+        self._world_base_fixed_joint_prim.GetAttribute("physics:localRot0").Set(
+            lazy.pxr.Gf.Quatf(*orientation[[3, 0, 1, 2]].tolist())
+        )
+
+    # Name of the actual root link that we are interested in. Note that this is different from self.root_link_name,
+    # which is "base_footprint_x", corresponding to the first of the 6 1DoF joints to control the base.
+    @property
+    def base_footprint_link_name(self):
+        return "base_link"
+
+    def _postprocess_control(self, control, control_type):
```

Comment:
we might consider making a `HolonomicBaseRobot` so that we can avoid duplication of the "magic base" logic going forward...but can leave for future PR, what do you think @cgokmen  @ChengshuLi ?

Reply:
I agree. Actually this can be done in 10 mins, right? Let's do it in this PR. The logic in `_post_load`, `_preprocess_control`, `control_limits` can be moved there easily.